song-server/src/main/java/bio/overture/song/server/service/AnalysisTypeService.java

```diff
@@ -244,11 +255,82 @@ private void validateAnalysisTypeSchema(JsonNode analysisTypeSchema) {
     }
   }
 
+  private JsonNode mergeSchema(JsonNode oldSchema, JsonNode newSchema) {
+    ObjectMapper mapper = new ObjectMapper();
+    ObjectNode mergedNode = mapper.createObjectNode();
+
+    // Iterate over all the fields in node1
+    oldSchema
+        .fieldNames()
+        .forEachRemaining(
+            fieldName -> {
+              JsonNode value1 = oldSchema.get(fieldName);
+              if (newSchema.has(fieldName)) {
+                JsonNode value2 = newSchema.get(fieldName);
+
+                // If both fields are objects, recursively merge them
+                if (value1.isObject() && value2.isObject()) {
+                  mergedNode.set(fieldName, mergeSchema(value1, value2));
+                } else {
+                  // If not both are objects, take the value from node2 (overwrite)
+                  mergedNode.set(fieldName, value2);
+                }
+              } else {
+                // If the field only exists in node1, add it as is
+                mergedNode.set(fieldName, value1);
+              }
+            });
+
+    // Add any fields from node2 that don't exist in node1
+    newSchema
+        .fieldNames()
+        .forEachRemaining(
+            fieldName -> {
+              if (!oldSchema.has(fieldName)) {
+                mergedNode.set(fieldName, newSchema.get(fieldName));
+              }
+            });
+
+    return mergedNode;
+  }
+
   @SneakyThrows
   private AnalysisType commitAnalysisType(
-      @NonNull String analysisTypeName, @NonNull JsonNode analysisTypeSchema) {
-    val analysisSchema =
-        AnalysisSchema.builder().name(analysisTypeName).schema(analysisTypeSchema).build();
+      @NonNull String analysisTypeName, @NonNull JsonNode newAnalysisTypeSchema, Options options) {
+
+    List<AnalysisSchema> schemaList = analysisSchemaRepository.findByName(analysisTypeName);
+    Optional<AnalysisSchema> oldSchemaOptional = Optional.empty();
+
+    if (CollectionUtils.isNotEmpty(schemaList)) {
+      oldSchemaOptional =
+          schemaList.stream()
+              .sorted(Comparator.comparing(AnalysisSchema::getCreatedAt).reversed())
+              .findFirst();
+    }
+    String fileTypes = "";
+
+    if (options != null && CollectionUtils.isNotEmpty(options.getFileTypes()))
+      fileTypes = String.join(",", options.getFileTypes());
```

Comment:
I would prefer if we use brackets to wrap all if statement bodies. This helps clarify what an if enables for future readers and prevents accidental bugs being introduced by future editors:

```suggestion
    if (options != null && CollectionUtils.isNotEmpty(options.getFileTypes())) {
      fileTypes = String.join(",", options.getFileTypes());
    }
```